names/slugs.go

```diff
@@ -199,3 +203,55 @@ func SanitizeLabelValues(labels map[string]string) {
 		labels[k] = ToValidLabelValue(v)
 	}
 }
+
+// StringToSlug receives any string and returns a human-friendly representation of it as a slug
+//
+// If the given inputs would produce an invalid slug, it returns an appropriate error
+func StringToSlug(str string) (string, error) {
+	// hash the string, take the first and last 4 characters of the hash
+	hashBytes := sha256.Sum256([]byte(str))
+	hashStr := hex.EncodeToString(hashBytes[:])
+	leadingDigest, trailingDigest := hashStr[:slugHashLength], hashStr[len(hashStr)-slugHashLength:]
+
+	// sanitize the string to be DNS Subdomain compatible
+	sanitizedStr, err := ToValidDNSSubdomainName(str)
+	if err != nil {
+		return "", err
+	}
+
+	// limit the length of the string so that the slug is not longer than the maximum DNS Subdomain length characters
+	const slugHashesLength = slugHashLength*2 + 2
+	const maxStrLength = maxDNSSubdomainLength - slugHashesLength
```

Comment:
question: if these constants are defined at compile time and do not depend on function inputs, why declare this inside a function?

Reply:
will move them :)